runtime/jvmti/suspendhelper.cpp

```diff
@@ -96,8 +96,8 @@ suspendThread(J9VMThread *currentThread, jthread thread, BOOLEAN allowNull, BOOL
 		if (VM_VMHelpers::isThreadSuspended(currentThread, threadObject)) {
 			rc = JVMTI_ERROR_THREAD_SUSPENDED;
 		} else {
-			U_64 internalSuspendState = (U_64)VM_VMHelpers::getCarrierVMThread(currentThread, threadObject);
-			J9OBJECT_U64_STORE(currentThread, threadObject, vm->internalSuspendStateOffset, (internalSuspendState | J9_VIRTUALTHREAD_INTERNAL_STATE_SUSPENDED));
+			J9VMThread *carrierVMThread = VM_VMHelpers::getCarrierVMThread(currentThread, threadObject);
+			J9OBJECT_U64_STORE(currentThread, threadObject, vm->internalSuspendStateOffset, ((U_64)carrierVMThread | J9_VIRTUALTHREAD_INTERNAL_STATE_SUSPENDED));
```

Comment:
in resumeThread, `NULL != carrierVMThread` is used to determine if we are in-transition. Will setting the `carrierThread` in `internalSuspendState` over here give a false indication that a thread is in-transition and lead to incorrect `vthreadInspectorCount += 1` operations? Over here, should we just do an OR operation to indicate the thread is suspended: `internalSuspendState | SUSPENDED_BIT`?

Reply:
Per discussion, will update this to use a direct `U_64` load and store macro to avoid confusion.